spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/parser/extensions/IcebergSqlExtensionsAstBuilder.scala

```diff
@@ -79,12 +81,33 @@ class IcebergSqlExtensionsAstBuilder(delegate: ParserInterface) extends IcebergS
   }
 
   /**
-   * Create a WRITE ORDERED BY logical command.
+   * Create a [[SetWriteDistributionAndOrdering]] for changing the write distribution and ordering.
    */
-  override def visitSetTableOrder(ctx: SetTableOrderContext): SetWriteOrder = withOrigin(ctx) {
-    SetWriteOrder(
-      typedVisit[Seq[String]](ctx.multipartIdentifier),
-      ctx.order.fields.asScala.map(typedVisit[(Term, SortDirection, NullOrder)]).toArray)
+  override def visitSetWriteDistributionAndOrdering(
+      ctx: SetWriteDistributionAndOrderingContext): SetWriteDistributionAndOrdering = {
+
+    val tableName = typedVisit[Seq[String]](ctx.multipartIdentifier)
+
+    val distributionSpec = ctx.writeDistributionSpec
+    val orderingSpec = ctx.writeOrderingSpec
+
+    if (distributionSpec == null && orderingSpec == null) {
+      throw new AnalysisException("Distribution and ordering spec cannot be empty at the same time")
```

Comment:
How about "ALTER TABLE has no changes: missing both distribute by and order by clauses"

Reply:
Will change.

Reply:
Done.